File: apps/web/src/lib/system.ts
export type BrokerState =
  | "NOT_CONFIGURED"
  | "UNKNOWN"
  | "RECONCILING"
  | "RECONCILED"
  | "DIVERGENT";

export interface SystemStatus {
  mode: "PAPER_ONLY";
  environment: string;
  broker_state: BrokerState;
  autonomous_execution_enabled: boolean;
  guardian_halted: boolean;
  guardian_reason: string | null;
}

export const h0SystemStatus: SystemStatus = {
  mode: "PAPER_ONLY",
  environment: "development",
  broker_state: "NOT_CONFIGURED",
  autonomous_execution_enabled: false,
  guardian_halted: true,
  guardian_reason: "H1 broker reconciliation has not been configured.",
};

File: packages/domain/system.py
from __future__ import annotations

from enum import StrEnum

from pydantic import BaseModel

from packages.configuration.settings import SystemMode


class BrokerState(StrEnum):
    NOT_CONFIGURED = "NOT_CONFIGURED"
    UNKNOWN = "UNKNOWN"
    RECONCILING = "RECONCILING"
    RECONCILED = "RECONCILED"
    DIVERGENT = "DIVERGENT"


class SystemStatus(BaseModel):
    mode: SystemMode
    environment: str
    broker_state: BrokerState
    autonomous_execution_enabled: bool
    guardian_halted: bool
    guardian_reason: str | None = None
